django/db/backends/oracle/base.py

```diff
@@ -259,6 +271,16 @@ def get_new_connection(self, conn_params):
 
     def init_connection_state(self):
         super().init_connection_state()
+
+        if Database.__name__ == "oracledb" and Database.is_thin_mode():
+            warnings.warn(
+                """
+                The support for the thin mode of oracledb is experimental. To switch
+                to the thick mode, please call oracledb.init_oracle_client() at the
+                beginning of your code.
+                """
+            )
```

Comment:
Why this warning is needed? We shouldn't raise a warning when this mode is accepted by Django, if it's not acceptable then we should always call `init_oracle_client()` (also without a warning).

Reply:
Get it. Warning removed.